server/routes/messageRoutes.js

```diff
@@ -8,7 +8,52 @@
 
 const express = require('express');
 const router = express.Router();
+const { validationResult } = require('express-validator/check');
+const { isLoggedIn, userIsMember } = require('./routeUtils');
+const message = require('../controllers/messagesController');
+const messageValidator = require('../controllers/validators/messageControllerValidator');
 
+router.post('/create/',
+    isLoggedIn,
+    userIsMember,
+    messageValidator.validate('checkMessage'),
+    function (req, res){
+        const errors = validationResult(req);
 
+        if (!errors.isEmpty()) {
+            res.status(202).json({ errors: errors.array()});
+        } else {
+            message.createMessage(req,req.user.uid)
+                .then( (message) => res.status(200).json({
+                    senderId: message.senderId,
+                    receiverId:message.receiverId,
+                    content: message.content,
+                }))
+                .catch(err => {
+                    res.status(500).json({ err: err.message });
+                });
+        }
+    }
+);
+
+router.get('/allUserMessages/',
```

Comment:
So remember that the full URL for this route is `http://localhost:3001/message/allUserMessages/`, we want to try and keep the routes as short and to the point as possible. Please rename this route to something like:
`router.get('/all/`)

Reply:
Addressed.